Observer/CustomerSaveAfterObserver.php

```diff
@@ -0,0 +1,41 @@
+<?php
+declare(strict_types=1);
+
+namespace MageOS\CommonAsyncEvents\Observer;
+
+use Magento\Customer\Model\Customer;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\MessageQueue\PublisherInterface;
+use Magento\Framework\Serialize\Serializer\Json;
+use MageOS\AsyncEvents\Helper\QueueMetadataInterface;
+
+class CustomerSaveAfterObserver implements ObserverInterface
+{
+    private Json $json;
+    private PublisherInterface $publisher;
+
+    public function __construct(
+        Json $json,
```

Comment:
you have a requirement for PHP 8.1, so you could you use property promotion here

Reply:
After speaking with Vinai, I'm lowering the PHP requirement to 7.4 instead.